### DISCUSSION TITLE
Changing fingerprints and Client Hints

### DISCUSSION THREAD

**Discussion by PatrickAlliander** [ultrafunkamsterdam/undetected-chromedriver#1106] [General]
I recently started using the undetected chromedriver and it works pretty great, including conbined with selenium-wire. By nature I'm not a python developer

As I've been diving deeper into browser detection and avoidance I've started seeing potentials caveats to catch bots, mainly in the area of fingerprinting and the client hints headers. I can easily tell Chrome to user a different user-agent, but the client hint headers it sends along easily give away that I am spoofing the user-agent with fake data. 

I think I can override the various the Sec-CH-UA headers by using the browser Options object like so. I just need to find a reasonably comprehensive list to select values from.
```
options = Options()
options.add_argument(f'--sec-ch-ua={sec_ch_ua}')
```

But how can I avoid fingerprinting? I've seen the [selenium-with-fingerprints plugin](https://github.com/CheshireCaat/selenium-with-fingerprints), but that's written in javascript. Is there a similar way this might be usable combined with undetected-chromedriver or is this unneccessary overengineering?

**Comment by AlexPaiva** [comment]
> I recently started using the undetected chromedriver and it works pretty great, including conbined with selenium-wire. By nature I'm not a python developer
> 
> As I've been diving deeper into browser detection and avoidance I've started seeing potentials caveats to catch bots, mainly in the area of fingerprinting and the client hints headers. I can easily tell Chrome to user a different user-agent, but the client hint headers it sends along easily give away that I am spoofing the user-agent with fake data.
> 
> I think I can override the various the Sec-CH-UA headers by using the browser Options object like so. I just need to find a reasonably comprehensive list to select values from.
> 
> ```
> options = Options()
> options.add_argument(f'--sec-ch-ua={sec_ch_ua}')
> ```
> 
> But how can I avoid fingerprinting? I've seen the [selenium-with-fingerprints plugin](https://github.com/CheshireCaat/selenium-with-fingerprints), but that's written in javascript. Is there a similar way this might be usable combined with undetected-chromedriver or is this unneccessary overengineering?

What exactly are you using on sec_ch_ua?

**Comment by ttlns** [comment]
@PatrickAlliander [Selenium-Profiles](https://github.com/kaliiiiiiiiii/Selenium-Profiles) might help you at that point

**Reply by cashmeredev** [reply]
searching for something like this for ages. thank you very much <3
